framework/src/modules/chain/logic/check_transaction_against_exceptions.js

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const exceptions = global.exceptions;
+
+const checkSenderPublicKeyException = transactionResponse => {
+	if (!exceptions.senderPublicKey.includes(transactionResponse.id)) {
+		return false;
+	}
+
+	if (!transactionResponse.errors.length > 1) {
+		return false;
+	}
+
+	if (transactionResponse.errors[0].dataPath !== '.senderPublicKey') {
+		return false;
+	}
+
+	return true;
+};
+
+const checkSignatures = (transactionResponse, transaction) => {
+	if (!exceptions.signatures.includes(transaction.id)) {
+		return false;
+	}
+
+	if (!transactionResponse.errors.length > 1) {
+		return false;
+	}
+
+	if (transactionResponse.errors[0].dataPath !== '.signature') {
```

Comment:
Are we sure it will always be the case that the first error will match the exception `.dataPath`? For this and the other checker functions in this file.

Reply:
Yes, we expect that the transactions will fail only with one error. That's why we have the previous `if` condition.